extractor/src/main/java/org/schabi/newpipe/extractor/InfoItem.java

```diff
@@ -68,7 +68,8 @@ public String toString() {
     public enum InfoType {
```

Comment:
Writing here, so a thread is created:

---

concerning your question https://github.com/TeamNewPipe/NewPipeExtractor/pull/788#issuecomment-1028713839

For me it It depends on what you want mixes to be. Are they a specialization of a playlist or are they a separate concept.  
I see NewPipe through the YouTube only lens, so for me it's a separate concept. A mix does not have a uploader or streamcount, which are the defining elements of playlists.

> maybe some services (will) have an uploader and/or a video count

Does any of the current services have it? If not do we plan on adding new services?

> backward compatibility

This should be the place that is affected: https://github.com/TeamNewPipe/NewPipeExtractor/blob/857b2f39e87353f4617706829dd42231e457a87e/extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfo.java#L631

Seeing as how it is not documented, which subtypes are going to be returned, one could hope that users of that functions already check for the type before using items of it. Like NewPipe itself does: https://github.com/TeamNewPipe/NewPipe/blob/9c9730b15217c7e8d3b2ccdecd658361215ccae3/app/src/main/java/org/schabi/newpipe/player/helper/PlayerHelper.java#L212  
Otherwise I find that this is an acceptable breaking change, since one can simply add such a check into their own app. It simply needs to be communicated properly. 
The alternative of adding a feature flag that is turned off by default, which toggles this [if statement](https://github.com/Stypox/NewPipeExtractor/blob/9f9c6cb49a6c4387958090d7112a880a7df8fb8a/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java#L623) I find very ugly which is why I'm willing to create the breaking change

Reply:
After thinking about it over the night, scrap most of what I said. Inside the app it will be displayed as a playlist and not something hugely different. So go with the playlist approach. 
A consideration is not using inheriting from `PlaylistInfoItem` but just simply adding the `mixType` field to it as `playlistType` with the type `NORMAL` and the ones already listed.
I can't think of any (dis-)advantes right now for that